### PR TITLE
[NEW] Smaller status messages

### LINKED ISSUE #906: More subtle join/leave messages
A small UX proposal. Right now, the join/leave messages are taking up a _ton_ of space in the room, making it hard to follow conversation over a longer period of time.

My proposal would be to make join/leave messages into smaller, 'inline' messages instead, like this:

selection_421

This would make it easier to 'scan' over them while following a conversation. On multiple subsequent joins/leaves, they could be grouped together with less margin inbetween, like this:

selection_422

The exact spacing inbetween entries would be up for discussion of course, but as a general proposal. I think it'd really help to focus more on the conversation, rather than on room state.
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/26967239-more-subtle-join-leave-messages?utm_campaign=plugin&utm_content=tracker%2F16890630&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F16890630&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by vynmera**
Closes #906

This PR adds an option to Preferences to collapse status messages such as joins and leaves. By default it is set to collapse them.

**Before (or when set to not collapse)**
2018-06-18!UNITO-UNDERSCORE!22-43-50

**After**
2018-06-18!UNITO-UNDERSCORE!22-40-29

**Option in Preferences**
2018-06-18!UNITO-UNDERSCORE!22-55-36
_Collapsed makes the joins and other status messages small. Full keeps them as they were before._

2018-06-18!UNITO-UNDERSCORE!22-56-26
It works for all status messages.

Enjoy! :heart:

### CODE REVIEW

**Comment by engelgabriel** [comment]
➤ Thiago Sanchez commented:

[bruno.quadros@rocket.chat](https://app.asana.com/0/653182561466463/list) Please, add here links for abstract with new layouts.

**Comment by engelgabriel** [comment]
➤ bruno.quadros@rocket.chat commented:

Abstract link: https://share.goabstract.com/9d13f28f-543d-4b6c-a51d-9528901debb3

**Comment by brunosquadros** [comment]
@vynmera Thanks so much for your PR! The idea is pretty good! We've adapted it a little bit visually to be more integrated with the other components in the interface. 

messeges-smaller-status

**Comment by vynmera** [comment]
@brunosquadros that's not really smaller lol, i like the icons but that's still huge

**Comment by thiagosanchz** [comment]
> @brunosquadros that's not really smaller lol, i like the icons but that's still huge

@vynmera 
We tried to create a solution to make the status less visible, not necessarily smaller.

**Comment by tassoevan** [comment]
@thiagosanchz @vynmera 
image

**Comment by brunosquadros** [comment]
We have tried some variations with Avatar image to be smaller but it was not fit very well with the look and feel that we have for the other components and we preferred did not change the size of the avatar.

**Comment by tassoevan** [comment]
@brunosquadros For your knowledge, there is a request to add reactions to some status messages: #4637

**Comment by engelgabriel** [comment]
This as closed by mistake by our integration with ASAN when the design team marked the design task as finished.

**Comment by engelgabriel** [comment]
compressed reply message
